cupy/core/_routines_sorting.pyx

```diff
@@ -266,7 +266,7 @@ def _partition_kernel(dtype):
     dtype = _get_typename(dtype)
     source = string.Template('''
     template<typename T>
-    __device__ void bitonic_sort_step(CArray<T, 1> a,
+    __device__ void bitonic_sort_step(CArray<T, 1, true> a,
```

Comment:
Is the input array always contiguous?

Reply:
Yes, sorry for not adding a comment.
Input is raveled here

https://github.com/cupy/cupy/blob/99e928685f15eecba38a1b1aac51c94d330cb6a0/cupy/core/_routines_sorting.pyx#L185